helpers/lib/resourceHints.js

```diff
@@ -0,0 +1,69 @@
+const utils = require("handlebars-utils");
+
+const resourceHintsLimit = 50;
+
+const defaultResourceHintState = 'preload';
+
+const resourceHintStates = [defaultResourceHintState, 'preconnect', 'prefetch'];
+
+const resourceHintFontType = 'font';
+const resourceHintStyleType = 'style';
+const resourceHintScriptType = 'script';
+const resourceHintAllowedTypes = [resourceHintStyleType, resourceHintFontType, resourceHintScriptType];
+
+/**
+ * @param {string} path - The uri to the resource.
+ * @param {string} state - 'preload' or 'preconnect'
+ * @param {string} type - any of [style, font, script]
+ */
+function addResourceHint(globals, path, state, type) {
+
+    if (typeof path !== 'string') {
+        throw new Error('resourceHint attribute require a valid URI');
+    }
+    path = path.trim();
```

Comment:
Do we need to `trim` the values (here and below)? That's basically mean that we allow whitespaces in the values